tests/integration/main.rs

```diff
@@ -0,0 +1,48 @@
+mod artichoke;
+
+use std::env;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+#[allow(dead_code)]
+#[derive(Debug)]
+struct CommandOutput<'a> {
+    call_args: Vec<&'a str>,
+    status: i32,
+    stdout: String,
+    stderr: String,
+}
+
+fn manifest_path() -> String {
+    env::var_os("CARGO_MANIFEST_DIR").unwrap().into_string().unwrap()
+}
+
+fn binary_path<'a>(name: &'a str) -> Result<PathBuf, String> {
+    let path = Path::new(&manifest_path()).join("target").join("debug").join(name);
+    match path.exists() {
+        true => Ok(path),
+        false => Err(format!("Can't find binary {} in ./target/debug/", name)),
+    }
+}
+
+fn run<'a>(binary_name: &'a str, call_args: Vec<&'a str>) -> Result<CommandOutput<'a>, String> {
+    let binary = binary_path(binary_name)?;
+
+    let output = Command::new(&binary)
+        .args(call_args.clone())
+        .output()
+        .expect(format!("Failed to run ruby app {}", binary_name).as_str());
```

Comment:
Since the execution will just throw things on stderr, I'm 99% sure this is a fatal error. e.g. Rust just failed spawn/execute entirely (I imagine this would never happen, but never say never I guess)

Reply:
sounds good, the `expect` makes sense here.